File: dayNine/python/mini_parking_functions.py
parking_space = [0] * 20

space_availability = [["Empty", "Parked"], ["Empty", "Parked"], ["Empty", "Parked"], ["Empty", "Parked"],["Empty", "Parked"], ["Empty", "Parked"], ["Empty", "Parked"], ["Empty", "Parked"],["Empty", "Parked"], ["Empty", "Parked"], ["Empty", "Parked"], ["Empty", "Parked"],["Empty", "Parked"], ["Empty", "Parked"], ["Empty", "Parked"], ["Empty", "Parked"],["Empty", "Parked"], ["Empty", "Parked"], ["Empty", "Parked"], ["Empty", "Parked"]]


def check_parking_space():
    slot_available = ["","","","","","","","","","","","","","","","","","","",""]
    for index in range(len(slot_available)):
        if parking_space[index] == 0:
            slot_available[index] = space_availability[index][0]
        if parking_space[index] == 1:
            slot_available[index] = space_availability[index][1]
    return slot_available


def choose_slot(slot):
    
    if parking_space[slot - 1] == 0:
        parking_space[slot - 1] += 1


def check_out(slot):
    
    if parking_space[slot - 1] == 1:
        parking_space[slot - 1] -= 1


def is_slot_empty(slot):
    is_empty = False
    if parking_space[slot - 1] == 0:
        is_empty = True
    return is_empty

File: dayNine/Java/MiniParkingFunctions.java
public class MiniParkingFunctions {
    private int [] parkingSpace;
    private String [][] spaceAvailability;
    
    public MiniParkingFunctions(){
        int [] parkingSpace = new int [20];
        String [][] spaceAvailability = {{"Empty","Parked"},{"Empty","Parked"},{"Empty","Parked"},{"Empty","Parked"},{"Empty","Parked"},{"Empty","Parked"},{"Empty","Parked"},{"Empty","Parked"},{"Empty","Parked"},{"Empty","Parked"},{"Empty","Parked"},{"Empty","Parked"},{"Empty","Parked"},{"Empty","Parked"},{"Empty","Parked"},{"Empty","Parked"},{"Empty","Parked"},{"Empty","Parked"},{"Empty","Parked"},{"Empty","Parked"}};
        
        this.parkingSpace = parkingSpace;
        this.spaceAvailability = spaceAvailability; 
    }
    
    
    public String[] checkParkingSpace(){
        String [] slotAvailable = new String[20];
        for(int index = 0; index < parkingSpace.length; index++){
            if(parkingSpace[index] == 0){
                slotAvailable[index] = spaceAvailability[index][0];
            }
            if(parkingSpace[index] == 1){
                slotAvailable[index] = spaceAvailability[index][1];
            }
        }
        
    return slotAvailable;
    }
    
    public void chooseSlot(int slot){
        if(parkingSpace[slot - 1] == 0){
            this.parkingSpace[slot - 1]++;
        }
    }
    
    public void checkOut(int slot){
        if(parkingSpace[slot - 1] == 1){
            this.parkingSpace[slot - 1]--;
        }
    }
    
    public boolean isSlotEmpty(int slot){
        boolean isEmpty = false;
        if(parkingSpace[slot - 1] == 0){
            isEmpty = true;
        }
    return isEmpty;
    }
}
